### ISSUE TITLE
Memory leak in v1.4.0 and newer

### ISSUE THREAD

**Issue by m417z** [pycompression/python-isal#166]
I'm using python-isal as a faster alternative to Python's gzip lib in [Winbindex](https://github.com/m417z/winbindex) deployment scripts. The data consists of a [large amount of compressed JSON files](https://github.com/m417z/winbindex/tree/gh-pages/data/by_filename_compressed). The deployment scripts run daily on Windows via GitHub Actions. A couple of days ago, the deployment started to fail with OOM errors. After various experiments, I pinpointed it to the new python-isal version.

Steps to reproduce:

* Download `windows.ui.xaml.dll.mui.json.gz` [from here](https://github.com/m417z/winbindex-data-insider/blob/f969340cce696e521482b3ce1dfe579743b1c15c/by_filename_compressed/windows.ui.xaml.dll.mui.json.gz).
* Place [this script](https://gist.github.com/m417z/1b43c2ecca911086af9cb62f8aaa8efa) in the same folder and run it.

Here's the output I'm getting on Windows, clearly showing the problem:
```
83312640 (79.5MiB)
116256768 (110.9MiB)
149090304 (142.2MiB)
181940224 (173.5MiB)
214781952 (204.8MiB)
247623680 (236.2MiB)
280465408 (267.5MiB)
313307136 (298.8MiB)
346148864 (330.1MiB)
378990592 (361.4MiB)
...
411889664 (392.8MiB)
444731392 (424.1MiB)
477577216 (455.5MiB)
510418944 (486.8MiB)
543260672 (518.1MiB)
576102400 (549.4MiB)
608944128 (580.7MiB)
641785856 (612.1MiB)
674627584 (643.4MiB)
707469312 (674.7MiB)
...
740311040 (706.0MiB)
773152768 (737.3MiB)
805994496 (768.7MiB)
838836224 (800.0MiB)
871677952 (831.3MiB)
904519680 (862.6MiB)
937361408 (893.9MiB)
970203136 (925.3MiB)
1003044864 (956.6MiB)
1035886592 (987.9MiB)
...
```

**Comment by rhpvorderman** [comment]
Hi, thank you for reporting this. For version 1.4.0  I have reimplemented the gzip reading component entirely in C. While this brings performance benefits, there are always potential oversights when implementing new code. 
Internally the readall implementation uses a list and `b"".join(chunk_list)` is used to produce the resulting decompressed result. Unfortunately I forgot to decrease the reference count for the list, resulting in a lot of objects remaning referenced.

**Comment by rhpvorderman** [comment]
The new 1.5.1 release should patch the issue. @m417z can you confirm?

I do not have my 2fa keys for PyPI on this computer, but once I get to the computer that does I will yank releases 1.4.0 to 1.5.0 so that users will not face this problem.

**Comment by m417z** [comment]
Seems to be fixed in 1.5.1, thanks for the prompt fix!

**Comment by rhpvorderman** [comment]
Thank you for reporting back! And also thanks for using python-isal!
